internal/sqld/sqldwebsockets/driver.go

```diff
@@ -0,0 +1,113 @@
+package sqldwebsockets
+
+import (
+	"context"
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"sort"
+)
+
+type sqldWsResult struct {
+	id      int64
+	changes int64
+}
+
+func (r *sqldWsResult) LastInsertId() (int64, error) {
+	return r.id, nil
+}
+
+func (r *sqldWsResult) RowsAffected() (int64, error) {
+	return r.changes, nil
+}
+
+type sqldWsRows struct {
+	res           *ExecResponse
+	currentRowIdx int
+}
+
+func (r *sqldWsRows) Columns() []string {
+	return r.res.Columns()
+}
+
+func (r *sqldWsRows) Close() error {
+	return nil
+}
+
+func (r *sqldWsRows) Next(dest []driver.Value) error {
+	if r.currentRowIdx == r.res.RowsCount() {
+		return io.EOF
+	}
+	count := r.res.RowLen(r.currentRowIdx)
+	for idx := 0; idx < count; idx++ {
+		v, err := r.res.Value(r.currentRowIdx, idx)
+		if err != nil {
+			return err
+		}
+		dest[idx] = v
+	}
+	r.currentRowIdx++
+	return nil
+}
+
+type sqldWsConn struct {
+	ws *SqldWebsocket
+}
+
+func SqldConnect(url string, jwt string) (*sqldWsConn, error) {
+	c, err := Connect(url, jwt)
+	if err != nil {
+		return nil, err
+	}
+	return &sqldWsConn{c}, nil
+}
+
+func (c *sqldWsConn) Prepare(query string) (driver.Stmt, error) {
+	return nil, fmt.Errorf("Prepare method not implemented")
+}
+
+func (c *sqldWsConn) Close() error {
+	return c.ws.Close()
+}
+
+func (c *sqldWsConn) Begin() (driver.Tx, error) {
+	return nil, fmt.Errorf("Begin method not implemented")
+}
+
+func convertArgs(args []driver.NamedValue) Params {
+	if len(args) == 0 {
+		return Params{}
+	}
+	sortedArgs := [](*driver.NamedValue){}
+	for idx := range args {
+		sortedArgs = append(sortedArgs, &args[idx])
+	}
+	sort.Slice(sortedArgs, func(i, j int) bool {
+		return sortedArgs[i].Ordinal < sortedArgs[j].Ordinal
+	})
```

Comment:
Can the `Ordinal` sequence have holes or duplicate values?

Reply:
I don't think so. The way I understand it is that Ordinal is just a position of the parameter.